src/Couchbase.Lite.Shared/API/Database/Collection.cs

```diff
@@ -881,12 +881,13 @@ internal void CheckCollectionValid()
             ThreadSafety.DoLocked(() =>
             {
                 if (c4Db == null) {
-                    throw new CouchbaseLiteException(C4ErrorCode.NotOpen, CouchbaseLiteErrorMessage.DBClosed);
+                    throw new CouchbaseLiteException(C4ErrorCode.NotOpen, CouchbaseLiteErrorMessage.DBClosedOrCollectionDeleted,
+                        new CouchbaseLiteException(C4ErrorCode.NotOpen, CouchbaseLiteErrorMessage.DBClosed));
                 }
 
                 if (_c4coll == IntPtr.Zero || !Native.c4coll_isValid((C4Collection*)_c4coll)) {
-                    throw new CouchbaseLiteException(C4ErrorCode.NotOpen, String.Format(CouchbaseLiteErrorMessage.CollectionNotAvailable,
-                                ToString()));
+                    throw new CouchbaseLiteException(C4ErrorCode.NotOpen, CouchbaseLiteErrorMessage.DBClosedOrCollectionDeleted,
+                        new CouchbaseLiteException(C4ErrorCode.NotOpen, String.Format(CouchbaseLiteErrorMessage.CollectionNotAvailable, ToString())));
```

Comment:
I know this looks a bit strange, but I really want to also log the actual cause of the exception. "Attempt to perform an operation on a closed database or a deleted collection." is a bit too general in my opinion.